config/src/test/scala/com/typesafe/config/impl/ConfigSubstitutionTest.scala

```diff
@@ -924,7 +924,7 @@ class ConfigSubstitutionTest extends TestUtils {
 
     @Test
     def substSelfReferenceIndirect() {
-        val obj = parseObject("""a=1, b=${a}, a=${b}""")
+        val obj = parseObject("""b=${a}, a=${b}""")
```

Comment:
They both appear to parse the object as `SimpleConfigObject({"a":1,"a":${b},"b":${a}})`, so I think the problem must be somewhere in `toConfig` or a method it calls.